server/oauth2.go

```diff
@@ -572,12 +588,15 @@ func (s *Server) validateCrossClientTrust(clientID, peerID string) (trusted bool
 }
 
 func validateRedirectURI(client storage.Client, redirectURI string) bool {
-	if !client.Public {
-		for _, uri := range client.RedirectURIs {
-			if redirectURI == uri {
-				return true
-			}
+	// Allow named RedirectURIs for both public and non-public clients.
```

Comment:
@sagikazarmark our use case is to use PKCE for a web app that doesn't run on "localhost".
Therefore, we would like that `RedirectURIs` are also considered for public clients.

The previous code to allow any localhost URI, `redirectURIOOB` or `deviceCallbackURI` for public clients is still active.
An alternative could be to no longer allow these special URLs when `RedirectURIs` is configured.
I don't have a strong opinion which option is better.

Reply:
I think it might be better to add this change in a separate PR (as it is not PKCE-specific) just to get the PKCE feature merged faster

Reply:
@tkleczek OK, I just created a separate PR here: https://github.com/dexidp/dex/pull/1822